services/calculate.go

```diff
@@ -167,152 +98,155 @@ func GetResponseHeaderScore(url string) (totalScore int, XSSReportURL string, ma
 		return 0, "", 0, nil, err
 	}
 	var responseHeaderMap map[string]string
+	// Sending Head Request to URL
 	response, err := http.Head(url)
 	if err != nil {
 		return 0, "", 0, nil, err
 	}
 	responseHeaderMap = make(map[string]string)
+	// Constructing Response Header Map
 	for k, v := range response.Header {
 		value := strings.Join(v, ",")
 		responseHeaderMap[k] = value
 	}
-	totalScore = 0
-	var score = 0
-	maxScore = 0
```

Comment:
Good Job moving the logic into their own functions, if you find time please document what each score means, and our intent behind calculating the same. It would be of help if it was detailed like the document you had shared before. 

Please make sure to either put that into the Wiki or the Readme.

Reply:
Let this be an open issue for now. Will add the docs soon.